### PR TITLE
Wip/fallback removal

### PR DESCRIPTION
**Pull request by fabiosky**
*Issue #10 , if available:*

*Description of changes:*
1. Image conversion is now using the most performant available path (sse2 or avx2) even when the image width is not a multiple of 16 (sse2) or 32 (avx2)
2. Added code coverage through tarpaulin
3. Fixed bug in plane constraint computation
4. Check if the buffers contain enough data according to the desired conversion

Code was heavily refactored to keep into account for out of bounds errors. Previously this could result in undefined behaviour, since the library could access invalid data.

To ensure proper coverage, tarpaulin coverage was included. This was used also to fix 3. and 4.
Coverage reports now at 90%, but the actual coverage is even higher, because tarpaulin is not able to cover lines belonging to inlined functions or propagated constants.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 language: rust
+addons:
+    apt:
```
Comment (pbor) [inline .travis.yml:3]: two spaces indent?

File: src/lib.rs [hunk 2]
```diff
@@ -453,7 +453,8 @@ static mut GLOBAL_STATE: GlobalState = GlobalState {
 /// use dcv_color_primitives as dcp;
 /// dcp::initialize();
 /// ```
-pub fn initialize() {
+pub fn initialize(#[cfg(feature = "test_instruction_sets")] force_set: &str) {
```
Comment (pbor) [inline src/lib.rs:456]: This is a bit weird...

Shall we have a `intialize_with_instructionset` variant instead?

Comment (fabiosky) [reply src/lib.rs:456]: Yes. Seems more clean.

File: .travis.yml (hunk 1 above)
Comment (fabiosky) [reply .travis.yml:3]: Will fix this

**Comment by fabiosky** [comment]
Closed in favor of #47

**Pull request closed without merging**